<launch>
  <!-- Arguments -->
  <arg name="model" default="$(env TURTLEBOT3_MODEL)" doc="model type [burger, waffle, waffle_pi]"/>
  <arg name="cmd_vel_topic" default="/cmd_vel" />
  <arg name="odom_topic" default="odom" />
  <arg name="move_forward_only" default="false"/>

  <!-- move_base -->
  <node pkg="move_base" type="move_base" respawn="false" name="move_base" output="screen">
    <param name="base_local_planner" value="dwa_local_planner/DWAPlannerROS" />
    <param name="base_global_planner" value="ccd_planner/GlobalPlanner" />
    <rosparam file="$(find turtlebot3_navigation)/param/costmap_common_params_$(arg model).yaml" command="load" ns="global_costmap" />
    <rosparam file="$(find turtlebot3_navigation)/param/costmap_common_params_$(arg model).yaml" command="load" ns="local_costmap" />
    <rosparam file="$(find turtlebot3_navigation)/param/local_costmap_params.yaml" command="load" />
    <rosparam file="$(find turtlebot3_navigation)/param/global_costmap_params.yaml" command="load" />
    <rosparam file="$(find turtlebot3_navigation)/param/move_base_params.yaml" command="load" />
    <rosparam file="$(find turtlebot3_navigation)/param/dwa_local_planner_params_$(arg model).yaml" command="load" />
    <remap from="cmd_vel" to="$(arg cmd_vel_topic)"/>
    <remap from="odom" to="$(arg odom_topic)"/>
    <param name="alpha" value="0.3" />
    <param name="radius" value="0.1" />
    <param name="resolution" value="0.05" />
    <param name="planner_frequency" value="0.5" />
    <param name="lethal_threshold" value="200" />
  </node>

  <!-- coverage_progress -->
  <node pkg="full_coverage_path_planner" type="coverage_progress" name="coverage_progress" output="screen">
    <param name="coverage_radius" type="double" value="0.1" />
    <param name="coverage_effectivity" type="double" value="100" />
    <!-- <rosparam file="$(find ccd_planner)/maps/map.yaml"/> -->
  </node>
</launch>
